avocado/utils/process.py

```diff
@@ -151,6 +152,41 @@ def get_parent_pid(pid):
         return int(parent_pid)
 
 
+def _get_pid_from_proc_pid_stat(proc_path):
+    match = re.match(r'\/proc\/([0-9]+)\/.*', proc_path)
+    if match is not None:
+        return int(match.group(1))
+
+
+def get_children_pids(parent_pid, recursive=False):
+    """
+    Returns the children PIDs for the given process
+
+    TODO: this is currently Linux specific, and needs to implement
+    similar features for other platforms.
```

Comment:
That brings us to that discussion we had a long while ago - Is `psutil` still out of consideration? If not, we could add a dependency on it, since it gives most of what we need, with support for a very decent amount of platforms.

Reply:
I'm all in for using other modules whenever possible.  The problem with `psutil` is that its installation requires a C compiler and some headers.  That severely hurts the easy deployment of Avocado.